compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -221,6 +225,84 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 
+    fn compute_2229_migrations(&self, closure_def_id: DefId, closure_span: Span) {
+        let upvars = if let Some(upvars) = self.tcx.upvars_mentioned(closure_def_id) {
+            upvars
+        } else {
+            return;
+        };
+
+        let mut need_migrations = Vec::new();
+
+        let typeck_results = self.typeck_results.borrow();
+        let min_captures = typeck_results.closure_min_captures.get(&closure_def_id);
+
+        for (&var_hir_id, _) in upvars.iter() {
+            let ty = self.node_ty(var_hir_id);
+            let ty = self.infcx.shallow_resolve(ty);
+            if ty.is_ty_infer() {
+                let span = self.tcx.hir().span(var_hir_id);
+                self.tcx.sess.struct_span_err(span, &format!("Infer")).emit();
+                println!("{:#?}", min_captures.unwrap()[&var_hir_id]);
+            }
+
+            if !ty.needs_drop(self.tcx, self.tcx.param_env(closure_def_id.expect_local())) {
```

Comment:
This fails for `error: internal compiler error: compiler/rustc_middle/src/ich/impls_ty.rs:167:9: ty::TyKind::hash_stable() - can't hash a TyVid _#39t.`